Game behaves very weird

I am on windows and tried both compiling myself from here and using the official download release.
Both behave very weird when I try to join a server, no matter if it is the official server `play.konstructs.org` or my own local server. What actually happens:

- I connect to the server
- I register a new account (or login to an account could should exist, not sure)
- The chat/console says "Connected to ..." and there could happen 2 things (imo randomly)

1. The game crashes instantly OR
2. I only see the skybox (in which I can rotate :D)

Of cause I started it in gdb already, but I can't happen to get a stacktrace. When it crashes, it only says `recv: No error`
The server throws out this bunch of text when the client crashes. No idea if it matters, but well...

```
[INFO] [07/18/2015 22:37:46.625] [main-akka.actor.default-dispatcher-5] [akka://
main/user/plugin-loader/server/$a] Message [akka.io.TcpPipelineHandler$Init$Even
t] from Actor[akka://main/user/plugin-loader/server/$b#-732856434] to Actor[akka
://main/user/plugin-loader/server/$a#-528818464] was not delivered. [1] dead let
ters encountered. This logging can be turned off or adjusted with configuration
settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [07/18/2015 22:37:46.625] [main-akka.actor.default-dispatcher-5] [akka://
main/user/plugin-loader/server/$a] Message [akka.io.TcpPipelineHandler$Init$Even
t] from Actor[akka://main/user/plugin-loader/server/$b#-732856434] to Actor[akka
://main/user/plugin-loader/server/$a#-528818464] was not delivered. [2] dead let
ters encountered. This logging can be turned off or adjusted with configuration
settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [07/18/2015 22:37:46.625] [main-akka.actor.default-dispatcher-5] [akka://
main/user/plugin-loader/server/$a] Message [akka.io.TcpPipelineHandler$Init$Even
t] from Actor[akka://main/user/plugin-loader/server/$b#-732856434] to Actor[akka
://main/user/plugin-loader/server/$a#-528818464] was not delivered. [3] dead let
ters encountered. This logging can be turned off or adjusted with configuration
settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [07/18/2015 22:37:46.679] [main-akka.actor.default-dispatcher-7] [akka://
main/user/plugin-loader/server/$a] Message [akka.io.Tcp$Aborted$] from Actor[akk
a://main/user/plugin-loader/server/$b#-732856434] to Actor[akka://main/user/plug
in-loader/server/$a#-528818464] was not delivered. [4] dead letters encountered.
 This logging can be turned off or adjusted with configuration settings 'akka.lo
g-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
[INFO] [07/18/2015 22:37:47.742] [main-akka.actor.default-dispatcher-2] [akka://
main/user/plugin-loader/universe/player-0] Message [konstructs.PlayerActor$Store
Data$] from Actor[akka://main/deadLetters] to Actor[akka://main/user/plugin-load
er/universe/player-0#-194649793] was not delivered. [5] dead letters encountered
. This logging can be turned off or adjusted with configuration settings 'akka.l
og-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
```

I just tried that out that minute.
Running the client of your link didn't change anything, but merging the latest commits was a catch! Compiled that one, and now I can join... no wait, it crashed again. But I had been on the server for rather a minute :o 
The crash occurred when I pressed `e` and messed around with my mouse wheel. No idea weather that matters.
And huh, now the game attends to crash every time when joining? (Yeah, I actually can join and see the world generating for a second).
Er, one thing to add: When I login to my previously created account `useus, paseus` I can see the world loading, but when I create a new account with any random name and pass, it crashes instantly again, without seeing any block of stone :)

PS: Is there a possibility to properly debug it or am I just to dumb/windows to find it? I mean with getting a callstack and the code position on crash, pausing it and messing around with internal variables etc...

PPS: my compiler throws out that a lot of the following warnings, looks like another bunch of dead code? Or my mistake again?
```
c:\mingw\include\inttypes.h:268:11: warning: inline function 'llabs' declared bu
t never defined [enabled by default]
 long long llabs (long long);
           ^
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,7 @@ endif()
 #-----------------------------------------------------------------------
 option(NANOGUI_BUILD_PYTHON OFF)
 option(NANOGUI_BUILD_EXAMPLE OFF)
+option(NANOGUI_BUILD_SHARED OFF)
 add_subdirectory(dependencies/nanogui)
 
 #-----------------------------------------------------------------------
@@ -49,9 +50,13 @@ FILE(GLOB SOURCE_FILES src/*.cpp dependencies/lodepng/lodepng.cpp src/noise/*.cp
 add_executable(konstructs ${SOURCE_FILES})
 
 if(MINGW)
-  target_link_libraries(konstructs ws2_32.lib
-    ${ZLIB_LIBRARIES} nanogui)
+  target_link_libraries(konstructs
+    ws2_32.lib
+    ${ZLIB_LIBRARIES}
+    nanogui ${NANOGUI_EXTRA_LIBS})
 else()
   target_link_libraries(konstructs
-    ${ZLIB_LIBRARIES} nanogui glfw)
+    ${ZLIB_LIBRARIES}
+    nanogui ${NANOGUI_EXTRA_LIBS}
+    glfw)
 endif()

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,6 +18,7 @@ endif()
 
 if(MINGW)
   add_definitions(-DWIN32)
+  add_definitions(-DEIGEN_DONT_ALIGN)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-pragmas")
 endif()
 

--- a/package/Makefile
+++ b/package/Makefile
@@ -221,6 +221,9 @@ docker-zip: /tmp/build-deps/zlib-bin
 	mkdir -p konstructs-client
 	cp build-windows/konstructs.exe konstructs-client/konstructs-client.exe
 	cp /tmp/build-deps/zlib-bin/bin/zlib1.dll konstructs-client
+	cp /usr/i686-w64-mingw32/sys-root/mingw/bin/libgcc_s_sjlj-1.dll konstructs-client
+	cp /usr/i686-w64-mingw32/sys-root/mingw/bin/libstdc++-6.dll konstructs-client
+	cp /usr/i686-w64-mingw32/sys-root/mingw/bin/libwinpthread-1.dll konstructs-client
 	cp -r ../textures konstructs-client
 	zip -r konstructs-client.zip konstructs-client
 	chown $$HOST_UID *.zip

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -527,6 +527,31 @@ private:
     nanogui::Window *window;
 };
 
+#ifdef WIN32
+int init_winsock() {
+    WORD wVersionRequested;
+    WSADATA wsaData;
+    int err;
+
+    wVersionRequested = MAKEWORD(2, 2);
+    err = WSAStartup(wVersionRequested, &wsaData);
+    if (err != 0) {
+        printf("WSAStartup failed with error: %d\n", err);
+        return 1;
+    }
+
+    if (LOBYTE(wsaData.wVersion) != 2 || HIBYTE(wsaData.wVersion) != 2) {
+        printf("Could not find a usable version of Winsock.dll\n");
+        WSACleanup();
+        return 1;
+    }
+
+    return 0;
+}
+#else
+int init_winsock() { return 0; }
+#endif
+
 void print_usage() {
     printf("OPTIONS: -h/--help                  - Show this help\n");
     printf("         -s/--server   <address>    - Server to enter\n");
@@ -571,6 +596,11 @@ int main(int argc, char ** argv) {
         //printf("Connecting to %s with user %s\n", server_addr, server_user);
     }
 
+    if (init_winsock()) {
+        printf("Failed to load winsock");
+        return 1;
+    }
+
     try {
         nanogui::init();
 
@@ -592,5 +622,8 @@ int main(int argc, char ** argv) {
         return -1;
     }
 
+#ifdef WIN32
+    WSACleanup();
+#endif
     return 0;
 }

